semgrep-core/parsing/Parse_kotlin_tree_sitter.ml

```diff
@@ -1344,13 +1345,25 @@ and nullable_type (env : env) ((v1, v2) : CST.nullable_type) =
     )
   in
   let v2 = List.map (token env) (* "?" *) v2 in
-  todo env (v1, v2)
+  (match v2 with
+   | hd::tl -> TyQuestion (v1, hd)
+   | _ ->
+       let q = Parse_info.fake_info "?" in
+       TyQuestion (v1, q)
+  )
 
 and parameter (env : env) ((v1, v2, v3) : CST.parameter) : parameter =
   let v1 = simple_identifier env v1 in
   let v2 = token env v2 (* ":" *) in
   let v3 = type_ env v3 in
-  todo env (v1, v2, v3)
+  let param = {
```

Comment:
You can factorize a bit and do let param = { (AST_generic.param_of_id v1) with ptype = v3 }